orangefs/tasks/main.yml

```diff
@@ -0,0 +1,87 @@
+---
+- name: update hosts to make sure names will resolve in hosts file
+  lineinfile:
+    dest: /etc/hosts
+    line: "{{ hostvars[item]['ansible_default_ipv4']['address'] }} {{ hostvars[item].inventory_hostname }}"
+    state: present
+  with_items: "{{ groups['orange-all'] }}"
+
+
+- name: Generate config file
+  command: pvfs2-genconfig --quiet --protocol tcp --ioservers "{{ groups['orange-data'] | join(',') }}" --metaservers "{{ groups['orange-data'] |join(',') }}" --storage /export/orangefs /etc/pvfs2.conf creates=/etc/pvfs2.conf
+  become: true
+
+- name: update storage path
+  lineinfile:
+    path: /etc/pvfs2.conf
+    regexp: 'DataStorageSpace /export/orangefs'
+    line: ' 	DataStorageSpace /export/orangefs/data'
+
+- name: update storage path
+  lineinfile:
+    path: /etc/pvfs2.conf
+    regexp: 'MetadataStorageSpace /export/orangefs'
+    line: ' 	MetadataStorageSpace /export/orangefs/meta'
+
+
+
+# Setup disks
+
+- name: install mdadm
+  dnf: name=mdadm state=latest
+  when: inventory_hostname in groups['orange-data']
+
+- name: setup mdraid 5 on nvmes
+  command: mdadm --create --verbose /dev/md0 --level=5 --raid-devices=4 /dev/nvme0n1 /dev/nvme1n1 /dev/nvme2n1 /dev/nvme3n1 creates=/dev/md0
```

Comment:
If /dev/md0 already exists, does this fail?  If it doesn't, it'll get flattened in the next task.  If it does, it means the playbook won't rerun.